torch/csrc/jit/passes/liveness.cpp

```diff
@@ -0,0 +1,85 @@
+#include <torch/csrc/jit/passes/alias_analysis.h>
+#include <torch/csrc/jit/passes/constant_pooling.h>
+#include <torch/csrc/jit/passes/liveness.h>
+#include <memory>
+
+namespace torch {
+namespace jit {
+
+struct LivenessAnalyzer {
+  explicit LivenessAnalyzer(std::shared_ptr<Graph> graph)
+      : graph_(std::move(graph)) {}
+
+  std::map<Node*, std::vector<Value*>> run() {
+    auto function_outputs = toSparseBitVector(graph_->block()->outputs());
```

Comment:
This is redundant, right? the first thing visiting a block does is union the outputs into the liveness.